### ISSUE TITLE
Fix for IEEE 754 conformance potentially incomplete

### ISSUE THREAD

**Issue by h-arlt** [jaxen-xpath/jaxen#108]
Hello,

Jaxen 1.1.6 was the 1st release that included a fix for JIRA issue JAXEN-227 (commit 4b67bbaa13c8814f33334f1104127cc0c25fa0a9).
Unfortunately, I cannot give any more details about this issue since the JIRA instance is not available any more.

Anyway, the commit mentioned above changed several classes dealing with numerical comparison but I guess that one has been missed: `org.jaxen.expr.DefaultGreaterThanEqualExpr`. It is still using `Double#compareTo(Double)` for comparison. 

Shouldn't it use Java comparison for `double` primitive values as `org.jaxen.expr.DefaultLessThanEqualExpr` does?

Thanks,
H

**Comment by elharo** [comment]
It's possible. Can you present a test case that would demonstrate this?

**Comment by h-arlt** [comment]
Unfortunately no, I cannot.

After spending some time with `org.jaxen.test.ArithmeticTest` and digging the docs (JLS, Java API Doc for `Double#compareTo(Double)` and IEEE 754), there are only two cases to consider:
1. `0 >= -0` (or `-0 >= 0` respectively)
2. `NaN >= x` where `x` is any Number

*NaN* is handled correctly by `org.jaxen.expr.DefaultRelationalExpr`. What's left is the comparison of zero and negative zero. But this is also determined correctly since the relation says *greater than or equal* and this is the only difference between `Double#compareTo(Double)` (treats 0 greater than -0) and Java comparison for the primitive type double (treats 0 equal to -0).

Long story short, it seems that my suspicion was purely based on the 'partially consistent' implementation (`DefaultGreaterThanEqualExpr` is the only class that uses `Double#compareTo(Double); all others fall back to Java comparison for double primitive values).

Hence, you might close this issue as there none. Of course, in case you want to re-align the implementation of `DefaultGreaterThanEqualExpr` to have a fully consistent implementation for comparing numbers - just in case anyone else stumbles across this - I won't object.

Thanks,
H
